### PR TITLE
Add ability to use Alt + Backspace to delete entire words in the console

### LINKED ISSUE #54: Alt + Backspace in the Console to delete entire words
The title should describe it pretty well I think. #58

### PR DESCRIPTION
**Pull request by MoonPadUSer**
Hold LALT and press backspace to delete an entire word in the console

Closes #54.

### CODE REVIEW

File: src/engine/console.cpp [hunk 1]
```diff
@@ -506,7 +507,21 @@ bool consolekey(int code, bool isdown)
 
             case SDLK_BACKSPACE:
             {
-                int len = (int)strlen(commandbuf), i = commandpos>=0 ? commandpos : len;
+                size_t commandbuf_len = strlen(commandbuf);
+                if (holding_alt)
```
Comment (TheAssassin) [inline src/engine/console.cpp:511]: Leave a blank line above, then insert a comment above the `if` explaining why this branch is added here.

File: src/engine/console.cpp [hunk 2]
```diff
@@ -506,7 +507,21 @@ bool consolekey(int code, bool isdown)
...
+                {
+                    char* last_word = strrchr(commandbuf, ' ');
```
Comment (TheAssassin) [inline src/engine/console.cpp:513]: Are you sure you're handling the empty string correctly?

File: src/engine/console.cpp [hunk 3]
```diff
@@ -506,7 +507,21 @@ bool consolekey(int code, bool isdown)
...
+                    {
+                        commandbuf[(commandbuf_len - strlen(last_word))] = 0;
+                    }
+                    else
+                    {
+                        commandbuf[0] = 0;
+                    }
+                }
+
+                int len = (int)commandbuf_len, i = commandpos>=0 ? commandpos : len;
```
Comment (TheAssassin) [inline src/engine/console.cpp:524]: Consider adding more blank lines to group what's logically related. I'd add one below here.

File: src/engine/console.cpp (hunk 2 above)
Comment (MoonPadUSer) [reply src/engine/console.cpp:514]: Not anymore, since you now said smth. Is there a better way to do this?

File: src/engine/console.cpp (hunk 3 above)
Comment (MoonPadUSer) [reply src/engine/console.cpp:526]: done.

File: src/engine/console.cpp (hunk 2 above)
Comment (TheAssassin) [reply src/engine/console.cpp:514]: Your code is not very complex, so try it in a separate `.cpp` file in an MWE. There you can call your code on different strings, and see if it crashes.

**Comment by TheAssassin** [comment]
Note for self: PR can and should be squashed.

File: src/engine/console.cpp (hunk 3 above)
Comment (TheAssassin) [reply src/engine/console.cpp:526]: Please feel free to just mark conversations as resolved when you're done.

File: src/engine/console.cpp (hunk 2 above)
Comment (MoonPadUSer) [reply src/engine/console.cpp:514]: Why should it crash tho, I can't see any reason why it would

File: src/engine/console.cpp [hunk 4]
```diff
@@ -506,7 +507,23 @@ bool consolekey(int code, bool isdown)
...
+                    {
+                        commandbuf[(commandbuf_len - strlen(last_word))] = 0;
+                    }
+                    else
+                    {
+                        commandbuf[0] = 0;
+                    }
+                }
+                
+
```
Comment (TheAssassin) [inline src/engine/console.cpp:525]: Now, there's two blank lines here, and none above the comment above the if...?

**Pull request merged**